src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -844,6 +848,25 @@ bool fuse_type_to_multiclass_nms(const std::shared_ptr<ov::Node>& node, const pr
     });
 }
 
+bool fuse_type_to_multinomial_v13(const std::shared_ptr<ov::Node>& node, const precisions_map& precisions) {
+    auto multinomial = ov::as_type_ptr<opset13::Multinomial>(node);
+    if (!multinomial) {
+        return false;
+    }
+
+    auto it_output_type = precisions.find(node->get_output_element_type(0));
+    if (it_output_type == precisions.end())
+        return false;
+
+    const auto& output_type = it_output_type->second;
+    if (output_type.is_integral_number() || output_type.is_real()) {
+        multinomial->set_convert_type(ov::element::i32);
+        multinomial->set_output_type(0, output_type, multinomial->get_output_partial_shape(0));
+        return true;
+    }
+    return false;
```

Comment:
You can consider using update_type function:

```suggestion
    return update_type(0, node, precisions, [&](const element::Type& to) {
        multinomial->set_convert_type(to);
    });
```